### PR TITLE
Add support for SMTP TLS reports

### PR DESCRIPTION
**Pull request by seanthegeek**
This is a draft PR. I need some help working on it.

### CODE REVIEW

**Comment by seanthegeek** [comment]
Ok. I think initial SMTP TLS support is done, including JSON, CSV, Elasticsearch, and Splunk output, but I haven't done much testing yet.

Other steps that are needed:

- Add SMTP TLS report output support to the Kafka client. @mikesiegel, can you look into this?
- Add SMTP TLS report output support to the LogAnalytics client. @rubeste, can you help with this?
- Add SMTP TLS report output support to the S3 client. @tom-henderson, can you help with that?
- Add SMTP TLS report output support to the Syslog client. @chris-y, can you help with this?
- Add SMTP TLS report dashboards for Splunk. I should be able help with that once I get more failure examples.
- Add SMTP TLS report to the Kibana dashboard. I haven't had ELK set up for a while, so if someone else could contribute dashboard changes would be awesome.
- Add SMTP TLS report support to the Grafana dashboard.
- Add SMTP TLS report samples and tests

**Comment by seanthegeek** [comment]
I was able to reuse existing code to create SMTP TLS report output support for Kafka, S3, Log Analytics, and Syslog. I have not tested any of these and testing would be appreciated.

**Comment by rubeste** [comment]
@seanthegeek I'll see if I can test it for Log Analytics. But I first need an SMTP TLS report.

File: parsedmarc/loganalytics.py [hunk 1]
```diff
@@ -161,3 +165,14 @@ def publish_results(
...
+        if (
+                results['smtp_tls_reports'] and
+                conf.dcr_smtp_tls_stream and
+                len(results['smtp_tls_reports']) > 0 and
+                save_smtp_tls):
+            logger.info("Publishing SMTP TLS reports.")
+            self.publish_json(
+                results['smtp_tls_reports'],
+                logs_client,
+                conf.dcr_forensic_stream)
```
Comment (rubeste) [inline parsedmarc/loganalytics.py:177]: Should be `conf.drc_smtp_tls_stream`.

**Comment by mikesiegel** [comment]
Hi @seanthegeek 

I haven't worked in a Kafka environment/dealing with DMARC since 2020. I can reach out to some old coworkers though and see if they can help out.

File: parsedmarc/loganalytics.py (hunk 1 above)
Comment (seanthegeek) [reply parsedmarc/loganalytics.py]: D'oh! Good catch! Fixed. Thanks!

**Comment by tom-henderson** [comment]
Hi @seanthegeek it's been a while since I've looked at this but the changes in s3.py look good to me 👍

**Comment by chris-y** [comment]
As far as I can tell the syslog looks ok, I'm not able to test though.

**Comment by rubeste** [comment]
@seanthegeek Created a Pull request #459 with some fixes and documentation.

**Comment by rubeste** [comment]
I have also tested the application and it works now.

**Pull request merged**